File: backend/MatrixMarket.Api/DTOs/ListingDtos.cs
namespace MatrixMarket.Api.DTOs;

public class CreateListingRequest
{
    public int CategoryId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public string ListingType { get; set; } = "Sell";
    public string? Isbn { get; set; }
    public string? ImageUrl { get; set; }
}

public class ListingResponse
{
    public int ListingId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public string ListingType { get; set; } = string.Empty;
    public string? Isbn { get; set; }
    public string? ImageUrl { get; set; }
    public string Status { get; set; } = string.Empty;
    public string CategoryName { get; set; } = string.Empty;
    public string SellerName { get; set; } = string.Empty;
    public int SellerId { get; set; }
    public int SellerKarma { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: android/app/src/main/java/com/matrixmarket/app/data/remote/dto/ListingDtos.kt
package com.matrixmarket.app.data.remote.dto

data class CategoryDto(
    val categoryId: Int,
    val categoryName: String,
    val description: String?
)

data class CreateListingRequest(
    val categoryId: Int,
    val title: String,
    val description: String,
    val price: Double,
    val listingType: String = "Sell",
    val isbn: String? = null,
    val imageUrl: String? = null
)

data class ListingResponse(
    val listingId: Int,
    val title: String,
    val description: String,
    val price: Double,
    val listingType: String,
    val isbn: String?,
    val imageUrl: String?,
    val status: String,
    val categoryName: String,
    val sellerName: String,
    val sellerId: Int,
    val sellerKarma: Int,
    val createdAt: String
)
